app/channels/program_channel.rb

```diff
@@ -30,19 +70,21 @@ def message(data)
         program.chars.destroy_all
       end
     end
-    ProgramChannel.broadcast_to(room, program.view)
+
+    ProgramChannel.broadcast_to(room, { action: :message, data: program.view })
   end
 
   def clear
-    program = Program.find(current_program.id)
+    program = current_program
     program.update(code: "")
     program.chars.destroy_all
-    ProgramChannel.broadcast_to(room, program.view)
+
+    ProgramChannel.broadcast_to(room, { action: :clear, data: program.view })
   end
 
   private
   def current_program
-    @current_program ||= Program.find(params.fetch(:id))
+    Program.find(params.fetch(:id))
```

Comment:
Just removed the memoization here for now due to the issue you pointed out. I _may_ still investigate doing some Rails caching with things like Program *if* it'll save us some bandwidth in the future.